src/order_cmd.cpp

```diff
@@ -4194,13 +4198,12 @@ CommandCost CmdBulkOrder(DoCommandFlags flags, const BulkOrderCmdData &cmd_data)
 			error_order.SetColour(COLOUR_RED);
 			error_order.SetLabelText("[Error] This order could not be parsed");
 
-			if (modify_pos != INVALID_VEH_ORDER_ID) {
-				DeleteOrder(v, modify_pos);
-				InsertOrder(v, std::move(error_order), modify_pos);
-				modify_pos = INVALID_VEH_ORDER_ID;
-			} else {
-				if (v->GetNumOrders() < MAX_VEH_ORDER_ID) InsertOrder(v, std::move(error_order), insert_pos);
+			if(modify_pos < get_import_end_index()){
+				CmdDeleteOrder(flags,v->index, modify_pos);
 			}
+			CmdInsertOrderIntl(flags,v, modify_pos, std::move(error_order),CIOIF_NONE);
+			modify_pos = get_import_end_index();
+			insert_pos = get_import_end_index();
```

Comment:
I am not very keen on this, as it sort of defeats the error handling mechanisms, and de-generalises the command. I'll make some adjustments later if I get time.

Reply:
Inserting in the middle has some unfortunate O(N^2) behaviour as well, I've got an idea for how this can be better sorted out, I'll make some changes later on.